src/porepy/models/fluid_mass_balance.py

```diff
@@ -743,6 +722,26 @@ def set_discretization_parameters(self) -> None:
                 },
             )
 
+    def set_boundary_data(self) -> None:
+        """Set values for the boundary conditions.
+
+        Both Neuman and Dirichlet data should be initialized here.
+
+        """
+        for bc_var_name in [self.bc_data_pressure_key, self.bc_data_darcy_flux_key]:
+            for bg, bg_data in self.mdg.boundaries(return_data=True):
+                bc_values = np.zeros(bg.num_cells)
```

Comment:
This needs to be a separate method. See how this is handled in poromechanics for time dependent mechanical conditions.

Reply:
The setter method could be generalized using a e.g. a list self._boundary_names: List[str]. The assignment using a "separate method" would then be

for name in self._boundary_names:
    for bg, bg_data in self.mdg.boundaries(return_data=True):
        bc_values = getattr(name+"_boundary_values")(bg)

This could cover all BCs for all physics.
Ask if clarification is needed!

Reply:
One thing worries me. How will this list work in a multiphysics case? Say, we have single phase flow mixin with its list, and a mechanics mixin with its list. How and where would these lists merge into one? Maybe it's better to make a common method, which is responsible just for setting the value to iterate and moving the previous value to state, and then call it by hand from each mixin for its variables?

Reply:
You'd merge them in the multi-physics combination.

Reply:
Closing after discussion in person.